addons/html_builder/static/src/core/building_blocks/builder_many2many.js

```diff
@@ -0,0 +1,117 @@
+import { Component, useComponent } from "@odoo/owl";
+import { useService, useAutofocus } from "@web/core/utils/hooks";
+import { debounce } from "@web/core/utils/timing";
+import {
+    basicContainerBuilderComponentProps,
+    getAllActionsAndOperations,
+    useBuilderComponent,
+    useDomState,
+} from "./utils";
+import { BuilderComponent } from "./builder_component";
+import { DropdownItem } from "@web/core/dropdown/dropdown_item";
+import { Dropdown } from "@web/core/dropdown/dropdown";
+
+export class BuilderMany2Many extends Component {
+    static template = "html_builder.BuilderMany2Many";
+    static props = {
+        ...basicContainerBuilderComponentProps,
+        model: String,
+        fields: { type: Array, element: String, optional: true },
+        domain: { type: Array, optional: true },
+        limit: Number,
+        id: { type: String, optional: true },
+        // currently always fakem2m
+        // currently always allowDelete
+    };
+    static components = { BuilderComponent, Dropdown, DropdownItem };
+
+    setup() {
+        this.orm = useService("orm");
+        useBuilderComponent();
+        const comp = useComponent();
+        this.comp = comp;
+        const { getAllActions, callOperation } = getAllActionsAndOperations(comp);
+        this.callOperation = callOperation;
+        this.applyOperation = comp.env.editor.shared.history.makePreviewableOperation(
+            this.callApply.bind(this)
+        );
+        this.searchResults = [];
+        this.state = useDomState(() => {
+            const el = this.env.getEditingElement();
+            const getAction = comp.env.editor.shared.builderActions.getAction;
+            const actionWithGetValue = getAllActions().find(
+                ({ actionId }) => getAction(actionId).getValue
+            );
+            const { actionId, actionParam } = actionWithGetValue;
+            const actionValue = getAction(actionId).getValue({
+                editingElement: el,
+                param: actionParam,
+            });
+            const selection = JSON.parse(actionValue || "[]");
+            return {
+                selection: selection,
+                searchResults: this.searchResults,
```

Comment:
searchResults, It s never update it during the useDomState. Maybe add it in another reactive or after. Or you want to reset it at each addStep ?
this.state.searchResults = this.searchResults